Central-Server/API/util/communication/packets.py

```diff
@@ -0,0 +1,500 @@
+# Contains all communication packets required for server-datastreamer communication.
+# Contains constructor functions for each packet and also a decode function for server packets.
+
+import json
+from enum import Enum
+import serial
+import util.communication.communication_interface as communication_interface
+from typing import List
+
+class PacketDeserializeException(Exception):
+    "Raised a packet fails to deserialize"
+    def __init__(self, packet_string, reason="Generic deserialization error", message="Failed to deserialize packet"):
+        self.packet_string = packet_string
+        self.message = message
+        self.reason = reason
+        if len(self.packet_string) > 100:
+            # If a packet string is over 100 characters, we truncate it
+            self.packet_string = self.packet_string[0:99] + ".. (rest hidden)"
+        super().__init__(self.message + " " + packet_string + " ==> " + self.reason)
+
+class DataPacketType(int, Enum):
+    """Enum describing the different types of packets we can send/recieve"""
+    IDENT = 0
+    """[CLIENT] Identification packet. Sent to servers to identify this as a testing rack."""
+    ID_CONFIRM = 1
+    """[CLIENT] Identification confirmation. Sent to a server after failure to confirm this rack's identity"""
+    READY = 2
+    """[CLIENT] Sent when this board is ready for a new job"""
+    DONE = 3
+    """[CLIENT] Sent when a board is done with a job"""
+    INVALID = 4
+    """[CLIENT] Sent in response to a [SERVER] packet that's invalid"""
+    BUSY = 5
+    """[CLIENT] Sent when the client is currently busy with setup or running, and cannot complete the request"""
+    JOB_UPDATE = 6
+    """[CLIENT] Sent intermittently while a job is running"""
+    PONG = 7
+    """[CLIENT] Sent in response to a PING packet"""
+    HEARTBEAT = 8
+    """[CLIENT] Sent to update the Kamaji server of this board's status"""
+    # Server packets
+    IDENT_PROBE = 100
+    """[SERVER] Requests the identity of a board. Expects an ID_CONFIRM response"""
+    PING = 101
```

Comment:
with heartbeats, are ping/pong useful, should we remove?

Reply:
Let's discuss at our next meeting, I think it may be useful to keep them just in case we find a use, but I'm willing to change my mind